features/notifications/notifications_controller.rb

```diff
@@ -0,0 +1,83 @@
+require_relative "../../shared/authenticated_controller_base"
+require_relative "../../services/services"
+require_relative "../../taskqueue/task_queue"
+require "pathname"
+
+module FastlaneCI
+  # CRUD controller for handling and displaying notifications to user
+  class NotificationsController < AuthenticatedControllerBase
+    HOME = "/notifications"
+
+    # Instantiates a new `Notification Controller` with a task queue to process
+    # requests
+    #
+    # @param  [Sinatra::Base] app
+    def initialize(app = nil)
+      super(app)
+      @task_queue = TaskQueue::TaskQueue.new(name: "notifications")
```

Comment:
Let's push this down one level into a service. I think it's possible that `@task_queue` could be garbage collected due to the lifecycle of `Sinatra::Base`
Generally speaking, we shouldn't be using `TaskQueue` in these controllers because of that. Also, any time we want a task queue in a controller, it might be a sign that we actually just want to call some async api on a server 😄